src/socket_poller.hpp

```diff
@@ -81,6 +81,21 @@ namespace zmq
         bool check_tag ();
 
     private:
+        void zero_trail_events(zmq::socket_poller_t::event_t *events_,
+                                                          int n_events_,
+                                                          int found);
+        int check_events(zmq::socket_poller_t::event_t *events_, int n_events_
```

Comment:
Maybe put the whole check_events definition within each #if branch. It is somewhat hard to read this way.

Reply:
@sigiesec Not against if there is consensus. So if someone else can do +1 or -1 on this, I will add it in the style refactor.